library/Icinga/File/Pdf.php

```diff
@@ -61,12 +62,29 @@ public function renderControllerAction($controller)
             'src="' . Icinga::app()->getBootstrapDirectory() . '/img/',
             $html
         );
+
+        $request = $controller->getRequest();
+
+        if (Hook::has('Pdfexport')) {
```

Comment:
The combination of Hook::has and (any, not first) `Hook->isSupported` could be moved to a static method in the base abstract Hook class - like `PdfExportHook::implementationIsAvailable()` or similar

Reply:
Again, this hook is only for our own purpose right now. I expect that there is exactly one hook which provides pdfexort. The dompdf rendering is stuff of our framework and will die once we've finished our pdfexport module. If there are more pdfexport hooks available the user would need a configuration option as well. Far out of scope at the moment.

Reply:
See below. Btw, I wouldn't make it die but move it to a dedicated module once deprecated - for those unable or unwilling to install Chromium for whatever reason. Compared to DomPDF it's a "monster" dependency. The configuration option is a good point for not leaving such logic in `File\Pdf`, I also had that in my mind but didn't want to clutter this comment with too many details.

Reply:
Come on, dompdf is a monster dep as well with all the libs it requires. You know all the bugs we have with it and that it's slow. It will die. Again, if there's a module other than our own in the future or if our module requires changes in this hook, we'll change it. For now, there's no point in doing so.

Reply:
Then there is no point for a hook, file_exists for chromium in a bunch of well-known dirs should suffice.